NativePrograms/PokemonSV/Programs/EggHatcher_Core.c

```diff
@@ -7,72 +7,77 @@
 
 void fly_to_gate(void) {
     pbf_press_button(BUTTON_A, 20, 500);
-    pbf_press_button(BUTTON_Y, 20, 800);
+    pbf_press_button(BUTTON_Y, 20, 1000);
     pbf_move_left_joystick(STICK_MAX, STICK_MAX, 6, 500);
     pbf_mash_button(BUTTON_A, 1000);
 };
 
 void move_to_location(void) {
     pbf_move_left_joystick(STICK_MIN, STICK_CENTER, 190, 0);
     pbf_press_button(BUTTON_L, 20, 0);
-    pbf_move_left_joystick(STICK_CENTER, STICK_MIN, 150, 100);
+    pbf_wait(200);
+    pbf_move_left_joystick(STICK_CENTER, STICK_MIN, 175, 200);
     pbf_move_left_joystick(STICK_CENTER, STICK_MAX, 20, 0);
     pbf_press_button(BUTTON_L, 20, 300);
 };
 
 void open_box(void) {
-    pbf_press_button(BUTTON_A, 20, 500);
-    pbf_press_button(BUTTON_X, 20, 200);
-    pbf_press_button(BUTTON_A, 20, 500);
+    pbf_press_button(BUTTON_A, 20, 700);
+    pbf_press_button(BUTTON_X, 20, 400);
+    pbf_press_button(BUTTON_A, 20, 700);
 };
 
 void get_egg(int boxCol) {
-    pbf_press_dpad(DPAD_RIGHT, 20, 100);
-    pbf_press_button(BUTTON_MINUS, 20, 100);
-    for (int i = 0; i < 40; i++) {
+    pbf_press_dpad(DPAD_RIGHT, 20, 400);
+    pbf_press_button(BUTTON_MINUS, 20, 400);
+    pbf_wait(400);
+    for (int i = 0; i < 60; i++) {

```

Comment:
It was there before but I do not get what this is doing. Why not multiply 5 by 60? Wouldn't it do the same ? And also allow to merge the wait at the end?

Reply:
Honestly same, I just saw that it was used to select the row of pokemon and it was working pretty well so didn't want to touch anything...I just increased its duration because in some instances 40 wasn't cutting it. I can try to just put the timer as 300 in the press dpad function. Though my hunch is they wanted to repeatedly press it instead of holding due to some weird interaction with the game.